### PR TITLE
Event handlers

### PR DESCRIPTION
**Pull request by gferrarocamus**

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
 
 class App extends React.Component {
-  render(){
+  state = {
+    total: null,
+    next: null,
+    operation: null,
+  };
+
+  handleClick = buttonName => {
```
Comment (rvvergara) [inline src/components/App.js]: React has a neat syntax for `setState`:

```javascript
handleClick = buttonName => this.setState(
    prevState => calculate(prevState, buttonName)
    );
```

Read more about it [here](https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/)

File: src/components/Button.js [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
-  const colorClass = `${props.color}-bg`;
-  const widthClass = props.wide ? 'wide' : '';
+const Button = ({
+  name, color, wide, clickHandler,
+}) => {
+  const colorClass = `${color}-bg`;
+  const widthClass = wide ? 'wide' : '';
   return (
-    <div className={`button ${colorClass} ${widthClass}`}>{props.name}</div>
+    <div className={`button ${colorClass} ${widthClass}`} onClick={clickHandler}>
```
Comment (rvvergara) [inline src/components/Button.js]: We can remove this linter error if we convert `div` into `button` and just handle the resulting smaller number fonts through css:

image

```javascript
return (
    <button
      type="button"
      className={`button ${colorClass} ${widthClass}`}
      onClick={clickHandler}
    >
      {name}
    </button>
  );
```

Please also note that if we have more than one `prop`/ attribute, we should write them in multiple lines as shown. Also pls note that `button` elements need `type` attributes for them.

File: src/components/Display.js [hunk 3]
```diff
@@ -1,22 +1,18 @@
...
+  </div>
+);
 
 Display.propTypes = {
-  result: PropTypes.string.isRequired,
-}
+  result: PropTypes.string,
+};
 
 Display.defaultProps = {
```
Comment (rvvergara) [inline src/components/Display.js]: We don't need `defaultProps` here anymore, we can just:
```javascript
Display.propTypes = {
  result: PropTypes.string.isRequired,
};
```

**Comment by dipto0321** [comment]
HI @gferrarocamus ,

So far so good. Successfully refactored the codes that mentioned by last feedback. This is from me 👍 . This project is ready to go.

Thanks 🙏 🙇

**Pull request merged**